### PR TITLE
MULE-11420: TypedValue and ParameterResolver can't be used inside a P…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…ojo annotated with @Nullsafe and Content

### CODE REVIEW

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java [hunk 1]
```diff
@@ -49,9 +52,11 @@ void runAndAssertTypedValue(String flowName, Object payloadValue, MediaType medi
   void assertTypedValue(TypedValue typedValue, Object payloadValue, MediaType mediaType, Charset charset) {
     assertThat(typedValue, is(instanceOf(TypedValue.class)));
     Object value = typedValue.getValue();
-    assertThat(value, is(instanceOf(payloadValue.getClass())));
     assertThat(value, is(payloadValue));
-    assertThat(typedValue.getDataType(), is(like(payloadValue.getClass(), mediaType, charset)));
+    if (value != null) {
```
Comment (alegmarra) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java]: why is it ok to have a null value? should this be two separate assertions?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ExpressionTypedValueValueResolver.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public ExpressionTypedValueValueResolver(String expression, Class<Object> expect
     } else {
       DataType expectedDataType =
           DataType.builder().type(expectedClass).mediaType(typedValue.getDataType().getMediaType()).build();
-      T result = (T) transform(typedValue, expectedDataType, event);
+      T result = typedValue.getValue() != null ? (T) transform(typedValue, expectedDataType, event) : null;
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ExpressionTypedValueValueResolver.java]: it is weird that you check `typedValue.getValue()` to not be null but then transform the entire `typedValue`. It sounds like this is a check that the transform method should do.

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java (hunk 1 above)
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java]: `null` is a valid resolution type, should be handled correctly.

Comment (alegmarra) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java]: yes, value can be null, but with this assertion we are not testing for expected null vs expected value

**Pull request merged**